packages/mco/components/data-policies/create-dr-policy/select-cluster-list.tsx

```diff
@@ -19,65 +21,106 @@ import {
   if (name)
     filteredClusters = filteredClusters.filter((c) => c.name.includes(name));
   return filteredClusters;
 };
 
+const fetchRegion = (cluster: ACMManagedClusterKind): string =>
+    cluster?.status?.clusterClaims?.reduce((region, claim) => region || (claim?.name === MANAGED_CLUSTER_REGION_CLAIM && claim?.value), "");
+
+const filterRegions = (filteredClusters: Cluster[]) => 
+  filteredClusters?.reduce((acc, cluster) => {
+    if (!acc.includes(cluster?.region) && cluster?.region !== "") {
+      acc.push(cluster?.region);
+    };
+    return acc;
+  }, []);
+
 export const SelectClusterList: React.FC<SelectClusterListProps> = ({
   selectedClusters,
   setSelectedClusters,
 }) => {
   const { t } = useTranslation('plugin__odf-console');
   const [isRegionOpen, setIsRegionOpen] = React.useState(false);
-  const [isZoneOpen, setIsZoneOpen] = React.useState(false);
   const [region, setRegion] = React.useState('');
-  const [zone, setZone] = React.useState('');
   const [nameSearch, setNameSearch] = React.useState('');
-  const [clusters] = React.useState<Cluster[]>([]);
+  const [clusters, setClusters] = React.useState<Cluster[]>([]);
+
+  const [acmManagedClusters, acmManagedClustersLoaded, acmManagedClustersLoadError] = useK8sWatchResource<ACMManagedClusterKind[]>({
+    kind: referenceForModel(ACMManagedClusterModel),
+    isList: true,
+    namespaced: false,
+  });
+
+  React.useEffect(() => {
+    if(acmManagedClustersLoaded && !acmManagedClustersLoadError) {
+      const getManagedClusterInfo = (cluster: ACMManagedClusterKind) =>(
+        {
+          "name": cluster?.metadata?.name,
+          "region": fetchRegion(cluster) ?? ""
+        }
+      );
+      setClusters(acmManagedClusters?.reduce((obj, acmManagedCluster) => ([...obj, getManagedClusterInfo(acmManagedCluster)]), []));
+    };
+  }, [acmManagedClusters, acmManagedClustersLoaded, acmManagedClustersLoadError]);
+
+  const filteredClusters: Cluster[] = React.useMemo(
+    () => getFilteredClusters(clusters, region, nameSearch),
+    [clusters, region, nameSearch]
+  );
 
   const onSelect: DataListCheckProps['onChange'] = (checked, event) => {
+    const name = filteredClusters?.[Number(event.currentTarget.id)]?.name;
+    const region = filteredClusters?.[Number(event.currentTarget.id)]?.region;
     if (checked)
-      setSelectedClusters([
+      setSelectedClusters((selectedClusters) => ({
         ...selectedClusters,
-        {
-          name: event.currentTarget.value,
-          region: event.currentTarget['data-region'],
+        [name]: {
+          name: name,
+          region: region,
+          storageClusterId: "",
+          storageSystem: "",
+          odfVersion: "",
         },
-      ]);
+      }));
     else {
-      const newClusters = selectedClusters.filter(
-        (c: Cluster) => c.name !== event.currentTarget.value
-      );
-      setSelectedClusters(newClusters);
+      let sc = {...selectedClusters};
+      delete sc?.[name]
```

Comment:
```suggestion
      let sc = _.cloneDeep(selectedClusters);
      delete sc?.[name]
```
Shallow copy is not enough if you want to work directly on the state object.